### PR TITLE
chore(deps): bump next major version to 14.x

### PR DESCRIPTION
**Pull request by direisc**
Next has an active vulnerability https://security.snyk.io/package/npm/next/13.5.4
Bump the next version to higher version.

I caught some issues running tests locally. If someone can help, I can try to solve the possible issues.

### CODE REVIEW

**Comment by direisc** [comment]
Experimental ESM Externals flags are set to `loose` to solve an issue when building admin UI.
This is related to an issue when building a `react-day-picker` peer dependency with no proper type (ESM/CommonJS).

NextJS shows some ways to solve it at https://nextjs.org/docs/messages/import-esm-externals
The workaround to don't need to fix the type of project or work at `react-day-picker` is adding `experimental: { esmExternals: 'loose' },` to let the Next build try to solve the import.

**Comment by direisc** [comment]
To clarify the issue around `react-day-picker`:

Screenshot 2024-06-12 at 12 34 55

File: packages/core/src/admin-ui/templates/next-config.ts [hunk 1]
```diff
@@ -1,5 +1,9 @@
 export const nextConfigTemplate = (basePath?: string) =>
   `const nextConfig = {
+    // Experimental ESM Externals
+    // https://nextjs.org/docs/messages/import-esm-externals
+    // required to fix build admin ui issues related to "react-day-picker" and "date-fn"
```
Comment (dcousens) [inline packages/core/src/admin-ui/templates/next-config.ts:5]: What issues did you hit?

**Comment by dcousens** [comment]
Thanks @direisc, this is really appreciated

**Pull request merged**